### PR TITLE
More refactoring of payoff.py and cooperation.py

### PR DESCRIPTION
**Pull request by meatballs**
Adds player_count function to derive number of players from interactions dictionary

### CODE REVIEW

**Comment by meatballs** [comment]
Don't merge this. I've changed my mind about the scores_matrix function. It means the payoff gets calculated twice and I don't like that! I'm going to get rid of it.

File: axelrod/tests/unit/test_payoff.py [hunk 1]
```diff
@@ -74,14 +80,23 @@ def setUpClass(cls):
             [-1.0, 0.0, 0.0, 1.0, 0.0, 0.0]
         ]
 
+    def test_player_count(self):
+        nplayers = ap.player_count(self.interactions)
+        self.assertEqual(nplayers, 3)
```
Comment (drvinceknight) [inline axelrod/tests/unit/test_payoff.py]: Worth including some edge cases? A single player etc...?

**Comment by meatballs** [comment]
done!

File: axelrod/tests/unit/test_payoff.py (hunk 1 above)
Comment (meatballs) [inline axelrod/tests/unit/test_payoff.py]: Yeah, probably! I'll think of something to add

**Comment by drvinceknight** [comment]
I'm happy with this. :+1:

**Comment by meatballs** [comment]
No excuses left now. I'm going to have tackle the ripping apart of Round Robin etc. to get the interactions out!

**Comment by drvinceknight** [comment]
:fearful: :muscle:

**Comment by marcharper** [comment]
looks good to me, @meatballs just checking that the don't merge comment above no longer applies?

**Comment by meatballs** [comment]
@marcharper correct. I removed my own code that was causing me offence!
